packages/components-react/icon-react/src/icon.tsx

```diff
@@ -0,0 +1,45 @@
+import type { HTMLAttributes } from 'react';
+import { clsx } from 'clsx';
+import { forwardRef } from 'react';
+
+type Dir = HTMLAttributes<HTMLSpanElement>['dir'];
+
+export type IconProps =
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden' | 'aria-label' | 'aria-labelledby' | 'dir'> & {
+      role?: never;
+      'aria-hidden'?: true;
+      'aria-label'?: never;
+      'aria-labelledby'?: never;
+      dir?: Dir;
+    })
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden'> & {
+      role: 'img';
+      'aria-label': string;
+      'aria-labelledby'?: never;
+      'aria-hidden'?: never;
+      dir?: Dir;
+    })
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden'> & {
+      role: 'img';
+      'aria-labelledby': string;
+      'aria-label'?: never;
+      'aria-hidden'?: never;
+      dir?: Dir;
+    });
```

Comment:
Kun je de `aria-*` regels in elk type het zelfde ordenen? Nu is het lastig zien waarin ze van elkaar verschillen

Reply:
Dat is precies het tegenovergestelde van wat ik wilde :D Ik dacht de focus te leggen op het verschil door daarmee te beginnen.